src/imagery/i.eodag/i.eodag.py

```diff
@@ -139,54 +166,110 @@ def get_bb(vector=None):
+            _(
+                "More than one area found in AOI map <{}>. \
+                      Using only the first area..."
+            ).format(vector)
         )
-    )
-    product, count = dag.search(id=query_id)
-    if count != 1:
-        raise ParameterError("Product couldn't be uniquely identified")
-    if not product[0].properties["id"].startswith(query_id):
-        raise ParameterError("Product wasn't found")
-    gs.verbose(
-        _("Poduct ending with: {} is found.".format(query_id[-min(len(query_id), 8) :]))
-    )
-    gs.verbose(_("Downloading..."))
-    dag.download(product[0])
 
+    # are we in LatLong location?
+    s = gs.read_command("g.proj", flags="j")
+    kv = gs.parse_key_val(s)
+    if "+proj" not in kv:
+        gs.fatal(_("Unable to get AOI: unprojected location not supported"))
 
-def download_by_ids(products_ids):
-    # Search in all recognized providers
-    for product_id in products_ids:
-        try:
-            download_by_id(product_id)
-        except ParameterError:
-            gs.error(
-                _(
-                    "Product ending with: {}, failed to download".format(
-                        product_id[-min(len(id), 8) :]
-                    )
-                )
-            )
+    geom_dict = gs.parse_command("v.out.ascii", format="wkt", **args)
+    num_vertices = len(str(geom_dict.keys()).split(","))
+    geom = [key for key in geom_dict][0]
+    if kv["+proj"] != "longlat":
+        gs.verbose(
+            _("Generating WKT from AOI map ({} vertices)...").format(num_vertices)
+        )
+        # TODO: Might need to check for number of coordinates
+        #       Make sure it won't cause problems like in:
+        #       https://github.com/OSGeo/grass-addons/blob/grass8/src/imagery/i.sentinel/i.sentinel.download/i.sentinel.download.py#L273
+        feature_type = geom[: geom.find("(")]
+        coords = geom.replace(feature_type + "((", "").replace("))", "").split(", ")
+        projected_geom = feature_type + "(("
+        coord_proj = Module(
+            "m.proj",
+            input="-",
+            flags="od",
+            stdin_="\n".join(coords),
+            stdout_=PIPE,
+            stderr_=PIPE,
+        )
+        projected_geom += (", ").join(
+            [
+                " ".join(poly_coords.split("|")[0:2])
+                for poly_coords in coord_proj.outputs["stdout"]
+                .value.strip()
+                .split("\n")
+            ]
+        ) + "))"
+        return projected_geom
+    return geom
+
+
+def search_by_ids(products_ids):
+    gs.message("Searching for products...")
+    search_result = []
+    for query_id in products_ids:
+        gs.message(_("Searching for {}".format(query_id)))
+        if options["provider"]:  # If provider is set, then search without fallback
+            product, count = dag.search(id=query_id, provider=options["provider"])
+        else:
+            product, count = dag.search(id=query_id)
```

Comment:
I guess this could be shortened:
```suggestion
        product, count = dag.search(id=query_id, provider=options["provider"] or None)
```
Then if no provider is given (options["provider"] equals ''), the provider keword get None